source/adiabatic_conditions/function.cc

```diff
@@ -62,6 +62,25 @@ namespace aspect
       return function.value(Point<1>(z), 0);
     }
 
+    template <int dim>
+    double Function<dim>::density (const Point<dim> &p) const
+    {
+      const double z = this->get_geometry_model().depth(p);
+      return function.value(Point<1>(z), 2);
+    }
+
+
+
+    template <int dim>
+    double Function<dim>::density_derivative (const Point<dim> &p) const
+    {
+      // TODO: better eps or make it a user input
```

Comment:
I am still not sure if we should require the user to specify this derivative. Thoughts?